### PR TITLE
style(sdk): improve type annotations and VSCode config for public API

### PR DESCRIPTION
**Pull request by speezepearson**
Description
-----------
- add type annotations to make my IDE better at go-to-definition and type inference
- add the dirs where our vendored GraphQL libs live to our .vscode Python config, so that VSCode can figure out what a `Client` is

Testing
-------
None -- these shouldn't affect any logic.

### CODE REVIEW

File: wandb/apis/public.py [hunk 1]
```diff
@@ -4852,14 +4890,14 @@ class ArtifactVersions(Paginator):
...
-        order=None,
-        per_page=50,
+        client: Client,
+        entity: str,
+        project: str,
+        collection_name: str,
+        type: str,
+        filters: Optional[Mapping[str, Any]] = None,
+        order: Optional[str] = None,
+        per_page: Optional[int] = 50,
```
Comment (kptkin) [inline wandb/apis/public.py:4900]: Is per_page Optional?

File: wandb/apis/public.py [hunk 2]
```diff
@@ -4946,7 +4984,13 @@ class ArtifactFiles(Paginator):
         % ARTIFACT_FILES_FRAGMENT
     )
 
-    def __init__(self, client, artifact, names=None, per_page=50):
+    def __init__(
+        self,
+        client: Client,
+        artifact: Artifact,
+        names: Optional[Sequence[str]] = None,
+        per_page: Optional[int] = 50,
```
Comment (kptkin) [inline wandb/apis/public.py:4992]: Same as above

**Review by kptkin — APPROVED** [review]
🙏

File: wandb/apis/public.py (hunk 1 above)
Comment (speezepearson) [reply wandb/apis/public.py]: Ehhh, technically yes in the GraphQL schema, but in practice I think I think it shouldn't be. Changed to `per_page: int`, thanks for calling this out!

**Pull request merged**